src/main/java/org/opentripplanner/middleware/bugsnag/BugsnagDispatcher.java

```diff
@@ -0,0 +1,217 @@
+package org.opentripplanner.middleware.bugsnag;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opentripplanner.middleware.bugsnag.response.Organization;
+import org.opentripplanner.middleware.models.BugsnagEvent;
+import org.opentripplanner.middleware.models.BugsnagEventRequest;
+import org.opentripplanner.middleware.models.BugsnagProject;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.JsonUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsInt;
+import static org.opentripplanner.middleware.spark.Main.getConfigPropertyAsText;
+
+/**
+ * Responsible for getting required information from Bugsnag. This is done by making calls to Bugsnag's API endpoints
+ * with a valid authorization token (BUGSNAG_API_KEY).
+ *
+ * A bugsnag API key is a key that is unique to an individual Bugsnag user. This key can be obtained by logging into
+ * Bugsnag (https://app.bugsnag.com), clicking on settings (top right hand corner) -> “My account settings”. From here
+ * select “Personal auth tokens” and then “Generate new token”.
+ *
+ * The following Bugsnag API endpoints are currently used:
+ *
+ * - https://api.bugsnag.com/organizations/<organization_id>/event_data_requests
+ * - https://api.bugsnag.com/organizations/<organization_id>/event_data_requests/<event_data_request_id>
+ * - https://api.bugsnag.com/organizations/<organization_id>/projects
+ */
+public class BugsnagDispatcher {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BugsnagDispatcher.class);
+
+    private static final String BUGSNAG_API_URL = "https://api.bugsnag.com";
+    private static final String BUGSNAG_API_KEY = getConfigPropertyAsText("BUGSNAG_API_KEY");
+    private static final String BUGSNAG_ORGANIZATION = getConfigPropertyAsText("BUGSNAG_ORGANIZATION");
+    private static final int BUGSNAG_REPORTING_WINDOW_IN_DAYS
+        = getConfigPropertyAsInt("BUGSNAG_REPORTING_WINDOW_IN_DAYS", 14);
+
+    /** Organization from which all projects and event request information will be extracted. Set once during
+     * startup. */
+    private static Organization ORGANIZATION = null;
+
+    /** Headers that are required by Bugsnag for each request. */
+    private static HashMap<String, String> BUGSNAG_HEADERS = new HashMap<>();
+
+    /** Filter object defining the boundaries on which event requests will be based. */
+    private static ObjectNode eventRequestFilter;
+
+    private static int CONNECTION_TIMEOUT_IN_SECONDS = 5;
+
+    static {
+        setBugsnagRequestHeaders();
+        buildEventRequestFilter();
+        setOrganization();
+    }
```

Comment:
We generally try to avoid static initialization where possible. It looks like most of these method calls would make more sense to just place directly in `makeEventDataRequest`. It's not clear to me what's gained by setting them here. As for `setOrganization`, I'm having trouble understanding why we even need the Organization object. Do we just get the ID from it? I would really appreciate more instructions in the PR description and in a README file with how to set this up so I can test on my own.

Reply:
PR description and README updated.

I've set these variables here because it seems the most efficient. The Bugsnag headers and organization ID are used in most calls to the Bugsnag API. Although the event request filter is only used in makeEventDataRequest, it makes sense to build it only once. Happy to move this into the method if you prefer?

It is not possible to get the organization id from the Bugsnag UI and it is needed for most API calls. The only way to get this is to make a call to the '/user/organizations/' API and extract the required organization based on the name given in the config file.